pip/commands/view.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+
+from pip.basecommand import Command
+from pip.commands.show import search_packages_info
+from pip.status_codes import SUCCESS, ERROR
+
+
+logger = logging.getLogger(__name__)
+
+
+class ViewCommand(Command):
+
+    """
+    Views the package source directory with the editor defined in
+    $PIP_EDITOR.
+    """
+    name = 'view'
+    usage = """
+      %prog <package>"""
+    summary = 'View installed package in the editor'
+
+    def __init__(self, *args, **kw):
+        super(ViewCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        if not args:
+            logger.warning('ERROR: Please provide a package name or names.')
+            return ERROR
+        if not os.getenv('PIP_EDITOR'):
+            logger.warning(
+                'ERROR: Please set $PIP_EDITOR to open the package.')
+            return ERROR
+        query = args
+
+        results = list(search_packages_info(query))
+        if len(results) is 0:
+            logger.warning("ERROR: Could not find package(s).")
+            return ERROR
+        for dist in results:
+            status_code = os.system("%s %s" % (os.getenv('PIP_EDITOR'),
+                                    dist['location'] + '/' + dist['name']))
```

Comment:
You should use subprocess here, and definitely should not invoke the editor via the shell, to avoid exploits...